dashboard/pages/fragment_graph.tsx

```diff
@@ -329,61 +287,82 @@ export default function Streaming() {
+
+  // Periodically fetch Prometheus back-pressure from Meta node
+  const { response: promethusMetrics } = useFetch(
+    fetchPrometheusBackPressure,
+    INTERVAL,
+    backPressureDataSource === "Prometheus"
+  )
+
+  // Periodically fetch embedded back-pressure from Meta node
+  // Didn't call `useFetch()` because the `setState` way is special.
+  const [embeddedBackPressureInfo, setEmbeddedBackPressureInfo] =
+    useState<EmbeddedBackPressureInfo>()
+  useEffect(() => {
+    if (backPressureDataSource === "Embedded") {
+      const interval = setInterval(() => {
+        fetchEmbeddedBackPressure().then(
+          (newBP) => {
+            console.log(newBP)
+            setEmbeddedBackPressureInfo((prev) =>
+              prev
+                ? {
+                    previous: prev.current,
+                    current: newBP,
+                  }
+                : {
+                    previous: newBP, // Use current value to show zero rate, but it's fine
+                    current: newBP,
+                  }
+            )
+          },
+          (e) => {
+            console.error(e)
+            toast(e, "error")
+          }
+        )
+      }, INTERVAL)
+      return () => {
+        clearInterval(interval)
+      }
+    }
+  }, [backPressureDataSource])
+
   const backPressures = useMemo(() => {
-    if (
-      (actorBackPressures && backPressureAlgo && backPressureAlgo) ||
-      backPressuresMetricsWithoutPromtheus
-    ) {
+    if (promethusMetrics || embeddedBackPressureInfo) {
       let map = new Map()
 
-      if (
-        backPressureDataSourceAlgo === "Embedded" &&
-        backPressuresMetricsWithoutPromtheus
-      ) {
-        for (const m of backPressuresMetricsWithoutPromtheus.outputBufferBlockingDuration) {
+      if (backPressureDataSource === "Embedded" && embeddedBackPressureInfo) {
+        const metrics = calculateBPRate(
+          embeddedBackPressureInfo.current,
+          embeddedBackPressureInfo.previous,
+          INTERVAL
+        )
+        // why sorting?
+        metrics.outputBufferBlockingDuration = sortBy(
+          metrics.outputBufferBlockingDuration,
+          (m) => (m.metric.fragmentId, m.metric.downstreamFragmentId)
+        )
+        for (const m of metrics.outputBufferBlockingDuration) {
           map.set(
             `${m.metric.fragmentId}_${m.metric.downstreamFragmentId}`,
             m.sample[0].value
           )
         }
-      } else if (
-        backPressureDataSourceAlgo !== "Embedded" &&
-        actorBackPressures
-      ) {
-        for (const m of actorBackPressures.outputBufferBlockingDuration) {
-          let algoFunc
-          switch (backPressureAlgo) {
-            case "p50":
-              algoFunc = p50
-              break
-            case "p90":
-              algoFunc = p90
-              break
-            case "p95":
-              algoFunc = p95
-              break
-            case "p99":
-              algoFunc = p99
```

Comment:
> Remove the p50/p90/p95/p99 selections in back-pressure page, which is actually not that thing in your mind

You mentioned this in PR description. So what is you want, the current back pressure rate rather than a percentile in past time range?

Reply:
It's a bit long story.

The percentiles didn't have a clear definition under this case. For example, the meaning of "the percentile of request latency" is clear and unambiguous: there are usually thousands of requests per seconds, and percentiles are the data points among these thousands of requests. However, in this cases, how to define percentiles of back-pressure? Among all actors, or among all the time? Either doesn't make sense to me.

In the existing implementation

1. Average among all actors (written in the PromQL in the Rust code)
2. Percentile among the data points during time range (written in web frontend)

I think most users including me don't expect this. After discussion with @BugenZhao, we decided to use average for both. Btw, actually, this part has already been merged in https://github.com/risingwavelabs/risingwave/pull/15380.